src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,310 @@
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"to", "quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            quitTo = data;
+        } else if (what == "to") {
+            to = data;
+        } else {
+            revert("RwaInputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Swaps the GEM balance of this contract into DAI through the PSM and push it into the `to` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function push() external isMate {
+        _doPush(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of GEM into DAI through the PSM and push it into the `to` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param amt Gem amount.
+     */
+    function push(uint256 amt) external isMate {
+        _doPush(amt);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out specific amount of GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param amt Gem amount.
+     */
+    function quit(uint256 amt) external isMate {
+        _doQuit(amt);
+    }
+
+    /**
+     * @notice Flushes out all outstanding `token` balance to `usr` address.
+     * @dev Can only be called by the admin
+     * @param token Destination address.
```

Comment:
This is the token address and not the destination address.
Also I think it would be nice to have the same exact documentation for `yank` in both conduits.

Reply:
Done